services/spar/test-integration/Util/Core.hs

```diff
@@ -288,13 +289,20 @@ nextWireId = Id <$> liftIO UUID.nextRandom
 nextSAMLID :: MonadIO m => m (ID a)
 nextSAMLID = ID . UUID.toText <$> liftIO UUID.nextRandom
 
+-- | Generate a 'SAML.UserRef' subject that should be fully acceptable by SAML and SCIM (i.e.
+-- the only requirement is that it's a valid email).
+nextSubject :: MonadIO m => m NameID
+nextSubject = do
+  email <- Brig.fromEmail <$> randomEmail
+  pure $ SAML.NameID (SAML.UNameIDEmail email) Nothing Nothing Nothing
+
 nextUserRef :: MonadIO m => m SAML.UserRef
 nextUserRef = liftIO $ do
-  (UUID.toText -> tenant) <- UUID.nextRandom
-  (UUID.toText -> subject) <- UUID.nextRandom
+  tenant <- UUID.toText <$> UUID.nextRandom
```

Comment:
I'm curious: is there a reason i'm missing why this is better?  (I don't care either way.)

Reply:
* The flow is broken (right-left-middle).

* It's significantly less common. I grepped and we _never_ do that in our codebase, other than in the two places above. (I mean, we use view patterns for sure, but in all those cases the pattern is more complicated, e.g. a nested tuple).